packages/web/components/trade-clipboard/index.tsx

```diff
@@ -0,0 +1,635 @@
+import { Currency } from "@keplr-wallet/types";
+import { CoinPretty, Dec, DecUtils } from "@keplr-wallet/unit";
+import { Pool } from "@osmosis-labs/pools";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import React, { FunctionComponent, useEffect, useRef } from "react";
+import { EmbedChainInfos } from "../../config";
+import {
+  useBooleanWithWindowEvent,
+  useSlippageConfig,
+  useTradeTokenInConfig,
+  useWindowSize,
+} from "../../hooks";
+import { useStore } from "../../stores";
+import { Error as ErrorBox } from "../alert";
+import { Button } from "../buttons";
+import { TokenSelect } from "../control/token-select";
+import { InputBox } from "../input";
+import { InfoTooltip } from "../tooltip";
+
+export const TradeClipboard: FunctionComponent<{
+  // Should be memorized
+  pools: Pool[];
+
+  containerClassName?: string;
+  containerStyle?: React.CSSProperties;
+  isInModal?: boolean;
+}> = observer(({ containerClassName, containerStyle, pools, isInModal }) => {
+  const {
+    chainStore,
+    accountStore,
+    queriesStore,
+    assetsStore: { nativeBalances, ibcBalances },
+    priceStore,
+  } = useStore();
+
+  const { isMobile } = useWindowSize();
+
+  const allTokenBalances = nativeBalances.concat(ibcBalances);
+
+  const account = accountStore.getAccount(chainStore.osmosis.chainId);
+  const queries = queriesStore.get(chainStore.osmosis.chainId);
+
+  const slippageConfig = useSlippageConfig();
+
+  const tradeTokenInConfig = useTradeTokenInConfig(
+    chainStore,
+    queriesStore,
+    chainStore.osmosis.chainId,
+    account.bech32Address,
+    undefined,
+    pools
+  );
+
+  const availableBalance = tradeTokenInConfig
+    ? queries.queryBalances
+        .getQueryBech32Address(account.bech32Address)
+        .getBalanceFromCurrency(tradeTokenInConfig.sendCurrency)
+    : undefined;
+
+  const showWarningSlippage = tradeTokenInConfig
+    ? slippageConfig.slippage
+        .toDec()
+        .lt(tradeTokenInConfig.expectedSwapResult.slippage.toDec()) ||
+      tradeTokenInConfig.expectedSwapResult.slippage.toDec().gt(new Dec(0.1))
+    : false;
+
+  const [isSettingOpen, setIsSettingOpen] = useBooleanWithWindowEvent(false);
+
+  const manualSlippageInputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isSettingOpen && slippageConfig.isManualSlippage) {
+      // Whenever the setting opened, give a focus to the input if the manual slippage setting mode is on.
+      manualSlippageInputRef.current?.focus();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isSettingOpen]);
+
+  const router = useRouter();
+  const firstQueryEffectChecker = useRef(false);
+
+  useEffect(() => {
```

Comment:
Let's create a `hooks/query-params` folder and put this logic in `useTokenQueryParams`